src/systems/a320_systems/src/pneumatic.rs

```diff
@@ -1,27 +1,3041 @@
+                1.,
+            )),
+            intermediate_pressure_valve: PurelyPneumaticValve::new(1.),
+            high_pressure_valve: ElectroPneumaticValve::new(1e-2, powered_by),
+            pressure_regulating_valve: ElectroPneumaticValve::new(1., powered_by),
+            fan_air_valve: ElectroPneumaticValve::new(1., powered_by),
+            transfer_pressure_pipe: PneumaticPipe::new(
+                Volume::new::<cubic_meter>(1.),
+                Pressure::new::<psi>(14.7),
+                ThermodynamicTemperature::new::<degree_celsius>(15.),
+            ),
+            precooler_inlet_pipe: PneumaticPipe::new(
+                Volume::new::<cubic_meter>(0.5),
+                Pressure::new::<psi>(14.7),
+                ThermodynamicTemperature::new::<degree_celsius>(15.),
+            ),
+            precooler_outlet_pipe: PneumaticPipe::new(
+                Volume::new::<cubic_meter>(0.5),
+                Pressure::new::<psi>(14.7),
+                ThermodynamicTemperature::new::<degree_celsius>(15.),
+            ),
+            precooler_supply_pipe: PneumaticPipe::new(
+                Volume::new::<cubic_meter>(1.),
+                Pressure::new::<psi>(14.7),
+                ThermodynamicTemperature::new::<degree_celsius>(15.),
+            ),
+            engine_starter_container: PneumaticPipe::new(
+                Volume::new::<cubic_meter>(0.5),
+                Pressure::new::<psi>(14.7),
+                ThermodynamicTemperature::new::<degree_celsius>(15.),
+            ),
+            engine_starter_exhaust: PneumaticExhaust::new(1e-2),
+            engine_starter_valve: DefaultValve::new_closed(),
+            precooler: Precooler::new(5.),
+        }
+    }
+
+    fn update(
+        &mut self,
+        context: &UpdateContext,
+        high_pressure_valve_controller: &impl ControllerSignal<HighPressureValveSignal>,
+        pressure_regulating_valve_controller: &impl ControllerSignal<PressureRegulatingValveSignal>,
+        engine_starter_valve_controller: &impl ControllerSignal<EngineStarterValveSignal>,
+        fan_air_valve_controller: &impl ControllerSignal<FanAirValveSignal>,
+        engine: &(impl EngineCorrectedN1 + EngineCorrectedN2),
+    ) {
+        // Update engines
+        self.fan_compression_chamber_controller
+            .update(context, engine);
+        self.intermediate_pressure_compression_chamber_controller
+            .update(context, engine);
+        self.high_pressure_compression_chamber_controller
+            .update(context, engine);
+
+        self.fan_compression_chamber
+            .update(&self.fan_compression_chamber_controller);
+        self.intermediate_pressure_compression_chamber
+            .update(&self.intermediate_pressure_compression_chamber_controller);
+        self.high_pressure_compression_chamber
+            .update(&self.high_pressure_compression_chamber_controller);
```

Comment:
**thought**: This is an interesting piece of code. I'm wondering if we should design the software such that the `Engine` itself owns these pieces, but maybe these pieces are provided into the `Engine`'s constructor function by the pneumatic system? Is this a perspective you have considered?

Reply:
I think it's a good idea to move the controllers' logic to the engine code. The idea will be to have the engines provide the target pressure and keep the `CompressionChamber` in the bleed system. That way, I can also get rid of some of the inappropriate `ControllerSignal` usage.

Reply:
It turns out there is quite a bit of work to this and I cannot really figure out a good solution, so I'll leave this as it is for now.